### PR TITLE
Feat cylc template

### PR DESCRIPTION
**Pull request by tdivoll**
This PR sets the framework to run the IceFloeTracker tasks in a Cylc pipeline. For now, we run the pipeline one, but using Cylc allows for cycling and more complicated workflows in the future.

### CODE REVIEW

**Comment by tdivoll** [comment]
I believe all the Register files were caught by the formatter.

File: scripts/pass_time_snakemake.py
Comment (cpaniaguam) [inline scripts/pass_time_snakemake.py:1]: If snakemake is not going to be used anymore, renaming this file might be sensible.

File: scripts/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+RUN apt-get install gdal-bin -y
+
+RUN pip3 install skyfield \ 
+        requests
+
+COPY fetchdata.sh /usr/local/bin
+COPY pass_time_snakemake.py /usr/local/bin
+
+ENTRYPOINT bash
```
Comment (cpaniaguam) [inline scripts/Dockerfile:17]:
```suggestion
ENTRYPOINT bash

```

File: cylc/flow.cylc [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+            $project_dir/scripts/ice-floe-tracker.jl landmask $fetchdata_dir $project_dir/$landmask_dir
+        """
+    [[preprocess]]
+        script = """
+            julia -t auto $project_dir/scripts/ice-floe-tracker.jl preprocess -t $fetchdata_dir/truecolor -r $fetchdata_dir/reflectance -l $project_dir/$landmask_dir -o $project_dir/$preprocess_dir
+         """
+    [[extractfeatures]]
+        script = """
+            julia -t auto $project_dir/scripts/ice-floe-tracker.jl extractfeatures -i $project_dir/$preprocess_dir -o $project_dir/$features_dir --minarea 300 --maxarea 90000
+        """
```
Comment (cpaniaguam) [inline cylc/flow.cylc:57]:
```suggestion
        """
        
```

**Review by cpaniaguam — CHANGES_REQUESTED** [review]
👍   Couple of comments about removing snakemake vestiges and dangling end-of-line at the end of non-,jl files.

**Review by broarr — APPROVED** [review]
Looks good to me! Most of these changes are just format changes right?

File: .github/workflows/build-docker.yml [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+
+      - name: Log in to the Container registry
+        uses: docker/login-action@v1
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
```
[the remark this answers is not in the document]
Comment (tdivoll) [reply .github/workflows/build-docker.yml:35]: I will play with that, it makes sense for software that will be open-sourced anyways

File: scripts/Dockerfile (hunk 1 above)
[the remark this answers is not in the document]
Comment (tdivoll) [reply scripts/Dockerfile:17]: That is correct! In the cylc pipeline I have one task each to call both of the scripts.

**Comment by tdivoll** [comment]
Replaced by #297 and by https://github.com/WilhelmusLab/ice-floe-tracker-pipeline/pull/13

**Pull request closed without merging**